src/Browscap/Helper/IteratorHelper.php

```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types = 1);
+namespace Browscap\Helper;
+
+use Psr\Log\LoggerInterface;
+
+class IteratorHelper
+{
+    /**
+     * @param LoggerInterface $logger
+     * @param string          $testKey
+     *
+     * @return array
+     */
+    public function getTestFiles(LoggerInterface $logger, string $testKey = 'full') : array
+    {
+        $data = [];
+
+        $checks          = [];
+        $sourceDirectory = __DIR__ . '/../../../tests/issues/';
+        $iterator        = new \RecursiveDirectoryIterator($sourceDirectory);
+
+        $errors = [];
+
+        foreach (new \RecursiveIteratorIterator($iterator) as $file) {
+            /** @var $file \SplFileInfo */
+            if (!$file->isFile() || 'php' !== $file->getExtension()) {
+                continue;
+            }
+
+            $tests = require $file->getPathname();
+
+            foreach ($tests as $key => $test) {
+                if (isset($data[$key])) {
+                    $logger->error('Test data is duplicated for key "' . $key . '"');
+                    $errors[] = 'Test data is duplicated for key "' . $key . '"';
```

Comment:
To reduce some of this duplication in the messages, I think it might be nice to set an `$error` variable or something like that and use that variable in both the `$logger->error` call and in the `$errors[]` push.

```php
$error = 'Test data is duplicated for key "' . $key . '"';
$logger->error($error);
$errors[] = $error;
```

Same for the two below as well.

Reply:
Also, I just realized that since the previous code was throwing an exception here, this doesn't behave the same way as it used to (I don't think that's a bad thing, just wanted to bring it up). Now if there are 5 cases of this particular error in the test suite, this will collect all 5 cases of it and store in the `$errors` array, and the methods in the test classes will output all 5 cases.  Previously, it would stop on the first encountered case.  This new approach may be more desirable though, what do you think?

Reply:
I removed the duplicate definition of error messages as you suggested.

Reply:
Its better now, I think. But I dont remember why I didnt just catch the exception in the new command.